drivers/CoreBattery/include/CoreBattery.h

```diff
@@ -15,10 +16,13 @@ namespace leka {
 class CoreBattery : public interface::Battery
 {
   public:
-	explicit CoreBattery(PinName voltage_pin, mbed::interface::DigitalIn &charge_status_input)
+	explicit CoreBattery(PinName voltage_pin, mbed::InterruptIn &charge_status_input)
 		: _voltage_pin {mbed::AnalogIn(voltage_pin, analog_voltage_reference)},
 		  _charge_status_input(charge_status_input) {};
 
+	void registerCallbackChargeInputOn(mbed::Callback<void()> const &callback);
+	void registerCallbackChargeInputOff(mbed::Callback<void()> const &callback);
```

Comment:
voici l'idée que j'ai eu par rapport aux noms de ces fonctions

```suggestion
	void onChargeDidStart(mbed::Callback<void()> const &callback);
	void onChargeDidStop(mbed::Callback<void()> const &callback);
```

ou alors

```suggestion
	void registerOnChargeDidStartCallback(mbed::Callback<void()> const &callback);
	void registerOnChargeDidStopCallback(mbed::Callback<void()> const &callback);
```

Apple/Swift utilise beaucoup le `did` ou le `will` pour définir si l'action a eu lieu ou va voir lieu, je trouve la nomenclature assez claire